client/my-sites/plans-features-main/index.jsx

```diff
@@ -184,7 +194,7 @@ export class PlansFeaturesMain extends Component {
 		}
 
 		if ( hideFreePlan ) {
-			plans.shift();
+			plans = plans.filter( plan => ! isFreePlan( plan ) );
```

Comment:
Continuing to comment on my own code, this would read a little clearer as `plans.filter( planSlug => ! isFreePlan( planSlug ) )`